File: baekjoon/js/10773.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const [, ...inputs] = buf.split("\n");

const stack = [];

for (const input of inputs) {
  const num = Number(input);

  if (num === 0) {
    stack.pop();
  } else {
    stack.push(num);
  }
}

const result = stack.reduce((acc, cur) => acc + cur, 0);

console.log(result);

File: baekjoon/rust/10773.rs
use std::io::stdin;

fn main() {
    let mut lines = stdin().lines().skip(1);

    let mut stack = Vec::new();

    while let Some(Ok(line)) = lines.next() {
        let num: u32 = line.trim().parse().unwrap();

        match num == 0 {
            true => {
                stack.pop();
            }
            false => stack.push(num),
        }
    }

    let result: u32 = stack.iter().sum();

    print!("{}", result);
}
